### ISSUE TITLE
Force type checking for empty abstract classes

### ISSUE THREAD

**Issue by fttx** [Microsoft/TypeScript#18249]
If i've these definitions:
```
export interface wsRequest {
    action: 'get' | 'put' | 'del';
    data: wsData;
}

export abstract class wsData { }

export interface wsDataGet extends wsData {
    path: string
}
```

and i do this:
```
let req: wsRequest = {
    action: 'putScanSessions',
    data: {anotherProperty: 'test'} // <---- here
};
```
it should warn me, because _anotherProperty_ is not preset in wsData neither in other interfaces that extends wsData

**Comment by kitsonk** [comment]
This is essentially a duplicate of #202

TypeScript is a [_structural_ typing system](https://github.com/Microsoft/TypeScript/wiki/FAQ#what-is-structural-typing), not a _nominal_ one.  Any class/structure/interface with no properties is assignable to any other subclass of that, because TypeScript is checking the structure.  This is well covered in the FAQs under various topics like [Why are all types assignable to empty interfaces?](https://github.com/Microsoft/TypeScript/wiki/FAQ#why-are-all-types-assignable-to-empty-interfaces).

**Comment by RyanCavanaugh** [comment]
Extra property checking does not trigger against empty types
